### ISSUE TITLE
Feature request: Give dashboard date filter the same options as question date filter

### ISSUE THREAD

**Issue by tomkanjam** [metabase/metabase#2959]
By default, all cards in my dashboard were using the "Past 7 days" filter.  

Since dashboard date filters cannot go beyond the card filter I need to specify a greater date range at the card level, ex: "This year", and set the default in the dashboard "Relative Date" filter to "Past 7 days".

The problem is that the user would like to drill down on the dashboard to specific days or ranges but that would require the addition of a second "Date Range" date filter to the dashboard. This gets messy very quickly.

It would be nice to have the same "fully featured" single dashboard date filter that questions do:

![image](https://cloud.githubusercontent.com/assets/303566/16827743/3e38abf0-493e-11e6-8092-15125cb3bf0a.png)

**Comment by mazameli** [comment]
Thanks for the feedback. One reason we split the time widget into these separate ones was an attempt to make dashboards and dashboard filtering dead simple and accessible for less frequent users of Metabase. I can see where you're coming from, but can you elaborate a bit on how adding a `specific date` filter along side your `relative date` filter "gets messy very quickly?"

**Comment by tomkanjam** [comment]
The user might wonder(as I did) which filter wins? 

Or they might forget that they set one filter and get bad information(as I did). 

Or they might be confused(as I was) by why the date filter they're playing with is not working.
In this case, my relative filter was set to "Past 7 days" but I was trying to use the range filter outside of the last 7 days so I was seeing data but not the data I was expecting. Not a big deal but also not a great experience having to manage two filters that do the same thing.

Another small gripe, if I'm jumping between two date filters I will usually want to clear the unused one(if I remember) and this causes all the cards to reload. I have to wait for everything to reload then proceed to set the new date filter and wait again. Depending on the speed of queries this can add a significant amount of wasted time.

Let me know if you need clarification on anything.

**Comment by mazameli** [comment]
That's really great feedback. Thanks for taking the time to write that out. We'll have to do some discussing about the best way to go with this, but I could envision one potential solution being to offer up the full datetime widget as an option when creating a time filter. A stopgap for now would be to encourage teammates to click through on a card in order to explore it beyond the bounds of what your dashboard filters permit.

**Comment by tomkanjam** [comment]
@mazameli cool, thanks for the response.

**Comment by tlrobinson** [comment]
@tomkanjan to clarify, do you want a date filter widget with all of the same options available in the query builder, or do you want dashboard date filters to override any date filters present on the original query, or both?

**Comment by tomkanjam** [comment]
@tlrobinson having both makes sense for my use case.

1. Dashboard date filter fully overrides card date filter
2. Dashboard date filter has the same options as the query builder date filter

**Comment by moumny** [comment]
The best would be to have to choice to show either one of 

- relative date
-single date
-full featured daterange picker

Depending on the use case.

**Comment by mazameli** [comment]
This feels like a no-brainer to me now, and would probably be trivial to implement (especially 'cause tlrobinson would be doing all the work, not me… ;) ).

Pick Time:
![1](https://cloud.githubusercontent.com/assets/2223916/22807181/7d521190-eeda-11e6-8074-cf080c775a37.png)

Choose "All Options:"
![2](https://cloud.githubusercontent.com/assets/2223916/22807228/c7f07bec-eeda-11e6-9925-abcfcef53232.png)

Behold it in all its splendrous glory:
![3](https://cloud.githubusercontent.com/assets/2223916/22807230/caac63c8-eeda-11e6-9900-f50db31d123a.png)

**Comment by tlrobinson** [comment]
Not quite trivial, but not hard either. Currently we use human-friendly and URL-safe date serializations, like

single: `2017-02-09`
range: `2017-02-09~2017-02-09`
month and year: `2017-03`
quarter and year: `Q1-2017`
relative: `past7days`

So we'd have to come up with serializations for the rest (or use MBQL in the URLs, which is kind of ugly for a user-facing string).

Here are some possibilities:

Previous: `pastNminutes` (hours, days, etc)
Current: `currentminute` (hour, day, etc) 
Next: `nextNminutes` (hours, days, etc)
Before: `~YYYY-MM-DD`
After: `YYYY-MM-DD~`
On: `YYYY-MM-DD`
Between: `YYYY-MM-DD~YYYY-MM-DD`

If we wanted to support time we'd also need to figure out how to incorporate that.

Also, I just realized our normal date picker doesn't even support arbitrary quarters or months.

**Comment by mazameli** [comment]
> Also, I just realized our normal date picker doesn't even support arbitrary quarters or months.

Yeah, you're right. The only workaround I can see is selecting specific dates from a calendar, but that doesn't let you select like each January for the past 3 years, or all Q2s, etc.

**Comment by moumny** [comment]
Would be very powerfull !
However i feel that the "old" filter in the query builder is both powerfull and easier to use.
By default you have the several options to select from (which is nice in dashboards) yet you can click on the "specific date" and select something there. 
I think its also more intuitive for new users.

**Comment by attekei** [comment]
Solved in #4607
